### ISSUE TITLE
Text expansion happens when trying to use qk edit and rm 

### ISSUE THREAD

**Issue by bostrt** [bostrt/quikey#36]
## Issue
When using `qk edit` or `qk rm` (or any future command that references the phrase name/key) you have to use quotes to prevent the text expansion from happening. For example:

```shell
$ qk edit -n :hello:<enter> #<-- this would trigger text expansion and mess up your edit session
$ qk edit -n ':hello:'<enter> #<-- this would work fine
```
Expecting users to always use quotations is a lot to ask and extra stuff to type; a little ironic for a text expansion app. 

This issue was initially brought up by @SanskritFritz in #19 
> when I type in the name (like qk rm --name=hello) it gets expanded but this time unwanted

## Workaround

As mentioned above, use quotes around the phrase name/key: `qk edit -n ':hello:'`.

## Better Solution

@SanskritFritz suggested using two identifiers for each phrase. One for management (edit, rm, etc) and another that triggers the text expansion. The naming of these identifiers should be obvious to the user and migration of a user's database from an old database to the new schema should be automatic. 

The new commands for adding and editing could look something like this:

```shell
$ qk add hello -t ":hello:" #<-- -t being for tag, the text you type to trigger expansion
$ qk edit hello
$ qk rm hello

$ qk add bye -t "//bye"
$ qk edit by
$ qk rm bye
```
Is it too much to ask user to remember two identifiers for the same phrase?

**Comment by bostrt** [comment]
@sdothum @xircon @joeynguyen @qalisto

You all have opened issues before for quikey. I'm curious if you have any thoughts about changing the command line as mentioned in the _**Better Solution**_ section in previous comment on this issue.

**Comment by SanskritFritz** [comment]
If backwards compatibilty is important (why not if possible and feasible), you could simply use --tag for editing and so on, since tags are already implemented:
```
$ qk add --name=',abc' --phrase='ABCDEFGHIJK' --tag=ABC
$ qk edit --tag=ABC --tag=ABCD # this adds a new tag ABCD
$ qk ls
+------+-------------------+----------------------------+---------------------------+
| Name | Tags              | Last Modified              | Phrase                    |
+------+-------------------+----------------------------+---------------------------+
| ,abc | ABC ABCD          | 2020-03-11T16:17:57.301163 | ABCDEFGHIJK               |
+------+-------------------+----------------------------+---------------------------+
$ qk edit --tag=ABCD --phrase='ABCDEFGHIJKLMN' # for simple cases why use an editor
$ qk rm --tag=ABC
```
Instead of --name we could use --trigger which would be more self explanatory:
```
$ qk add --trigger=',abc' --phrase='ABCDEFGHIJK' --tag='ABC'
```
May I also suggest another option for qk ls for easier scripting:
```
$ qk ls --list
Name;Tags;Phrase
",abc";"ABC ABCD";"ABCDEFGHIJK"
```

**Comment by bostrt** [comment]
Considering this is still a _very_ young project and few users, I'm less interested in keeping backwards compatibility and more interested in finding a clean solution for the future. 

[This post](https://jmmv.dev/2013/09/cli-design-subcommand-based-interfaces.html) has been a good reference for me. 

Considering this pattern:
```
tool [general options] command [command options] [command arguments]
```
The use of flags vs options in quikey right now doesn't make sense. Currently, `qk add -n '//test'` has the user specifying a *mandatory argument* as a option/flag. Same with `edit`, `rm`, etc. Why make users use flags for an argument that will always specify. 

Considering the advice in the article above and some classic commands with subcommand style: `git add <file>`, `git clone <repo>`, `kubectl logs <pod>`, `kubectl delete <rsrc>`, list goes on; I think moving to a better subcommand design would be great for quikey. Finding a viable migration from old databases to new is important to keep in mind as well. 

----

Another idea floating around is having a **configurable** prefix, postfix, or both that is attached to a phrase key. For example, a user may run:
```
$ qk add hello
hello has been added
Type "//hello" to execute!
```
and then utilize the new phrase using `"//hello"` if they have configured `"//"` as their special trigger keys. 
This would **prevent** the need for having 2 ID's to manage 1) an alias/name used for managing the phrases and 2) a trigger key phrase.

**Comment by SanskritFritz** [comment]
I like the first idea. So that would mean, we'd have a unique name for each entry. My thought was that by using tags we could have more such names, but that is probably too much flexibility although I would use that feature.

The second idea sounds very bad to me, I would have to define a non changing prefix for triggers, which is definitely not what I would want. Also we would be dependent on a config entry which very much hinders database portability. I have many kinds of triggers, some of them simply fix typos (I type "teh " and it gets fixed to "the ", wich is lovely).

**Comment by bostrt** [comment]
> I have many kinds of triggers, some of them simply fix typos (I type "teh " and it gets fixed to "the ", wich is lovely).

Okay, that's a use-case I certainly never considered and is a great idea. Forcing the use of a prefix would kill that use-case and I don't want to do that. You are right in that it takes us back to the first idea in this thread. I'll see if there's any other possibilities as well.

**Comment by bostrt** [comment]
I've been spending some time lately on other quikey tasks and wanted to update here too.

I'm still spinning on this change. The only thing I'm certain about right now is that I'll be cleaning up the CLI usage some per first half of https://github.com/bostrt/quikey/issues/36#issuecomment-598543024.

**Comment by xircon** [comment]
Sorry, I completely missed this.  I am happy with any format you are going to use.  
I use it for things like email addresses, shebangs and the like, so not extensive.

**Comment by bostrt** [comment]
I've had some other ideas offered to me about this issue that I'm going to go with to keep backwards compatibility and prevent the need for database schema changes (and migrations). 

Basically, prompt user when `qk edit` is run with no args but keeping the current `-n/--name` flag as an option. User will be shown a prompt with list of phrases where they can choose which one to edit.

I'll likely add this same logic to `qk rm`.

Thanks @rdmullett for the input :)

**Comment by SanskritFritz** [comment]
I oppose this idea because that prevents scripting or at least makes it very hard. Command line programs should keep interactions to a bara minimum IMO. I have nothing against it if it will be optional, thus we will be able to edit and rm without interactions (using Tags for example).

**Comment by bostrt** [comment]
Yeah, the non-interactive usage will definitely still remain.

**Comment by bostrt** [comment]
Closing this out per a123ddc which released in quikey 0.1.3 today.

**Comment by SanskritFritz** [comment]
Oh, wow, great news!
I'll lay my hands on the new version soon, and update the AUR package.
Can't wait to see how the new version works! I already watched the demo video :smile: 
Maybe updating the readme.md would be useful?

**Comment by xircon** [comment]
Where is the video?

**Comment by SanskritFritz** [comment]
> Where is the video?

https://blog.quikey.io/posts/release-0_1_3/

**Comment by xircon** [comment]
Thanks, very impressive !!

**Comment by bostrt** [comment]
>  Maybe updating the readme.md would be useful?

Yes! I'll add that to my todo list for this week:)

**Comment by bostrt** [comment]
>  I'll lay my hands on the new version soon, and update the AUR package.

I made a mistake :sob:  in the 0.1.3 release (accidentally shipped `test` module as quikey). This is corrected in the 0.1.4 release.
